rust/src/csv/reader.rs

```diff
@@ -259,7 +457,21 @@ mod tests {
 
         let file = File::open("test/data/null_test.csv").unwrap();
 
-        let mut csv = Reader::new(file, Arc::new(schema), true, 1024, None);
+        let mut csv = Reader::new(file, Some(Arc::new(schema)), true, None, None, 1024, None);
+        let batch = csv.next().unwrap().unwrap();
+
+        assert_eq!(false, batch.column(1).is_null(0));
+        assert_eq!(false, batch.column(1).is_null(1));
+        assert_eq!(true, batch.column(1).is_null(2));
+        assert_eq!(false, batch.column(1).is_null(3));
+        assert_eq!(false, batch.column(1).is_null(4));
+    }
+
+    #[test]
+    fn test_nulls_with_inference() {
```

Comment:
It's good to check for nulls but we should test the inference logic here too.  i.e. a column with ints and floats should be inferred to be float.  We should cover each of the cases possible.

Reply:
When inferring types we should not care if the values are quoted or not based on my interpretation of https://tools.ietf.org/html/rfc4180 which is as close to a specification as we have for CSV

Reply:
The RFC doesn't entirely address the concerns that I had here, and I think ignoring quoted values (if they're the only ones in otherwise non-quoted files) might lead to undesired/unintended differences. I'm mainly using Spark, Pandas and some SQL import tools that I've used to inform my gut-feel.

I'd like to do some research in the coming weeks when I'm back at work, to look at how various csv readers treat this case, and I can then update the code for that.

I propose opening a separate JIRA for this case ("Improve CSV Reader to allow specifying treatment of double quotes"), as I think it's valuable enough to spend a bit more time on ensuring that we conform with what's done in the wild. @andygrove @sunchao @paddyhoran  any opinions/disagreements?